geode-core/src/main/java/org/apache/geode/distributed/internal/LocatorLoadSnapshot.java

```diff
@@ -448,14 +486,28 @@ private void updateMap(Map map, ServerLocation location, float load, float loadP
     }
   }
 
+  private void updateMap(Map map, ServerLocation location, String memberId, float load,
+      float loadPerConnection) {
+    Map groupMap = (Map) map.get(null);
+    ServerLocationAndMemberId locationAndMemberId =
+        new ServerLocationAndMemberId(location, memberId);
+    LoadHolder holder =
+        (LoadHolder) groupMap.get(locationAndMemberId);
+
+    if (holder != null) {
+      holder.setLoad(load, loadPerConnection);
+    }
+  }
+
   /**
    *
    * @param groupServers the servers to consider
    * @param excludedServers servers to exclude
    * @param count how many you want. a negative number means all of them in order of best to worst
    * @return a list of best...worst server LoadHolders
    */
-  private List<LoadHolder> findBestServers(Map<ServerLocation, LoadHolder> groupServers,
+  private List<LoadHolder> findBestServers(
```

Comment:
Not a new method but significantly changed, it should be, at leat, unit tested.
You can make it package private, annotate it with `@VisibleForTesting` and access it directly from `LocatorLoadSnapshotJUnitTest` and/or `LocatorLoadSnapshotIntegrationTest`.

Reply:
done